### PR TITLE
Remove Draper

### PR DESCRIPTION
**Pull request by yez**
Finishes #102

These changes remove the use of Draper in favor of a built in Ruby
  mechanism: DelegateClass. This combined with leveraging Rails' view_context
  object help keep responsibilities separated from decorator and view rendering.

Draper was very invasive and as such, many specs had to change. Potentially
  controversial changes are evident in spec/decorators/event_decorator_spec.
  More specifically, specs have changed to expect messages onto the
  internal view_context for url and other html generation.

A helper file MarkdownHelper was deleted and its logic in-lined to the atom
  response block in the events controller.

Happy to explain/discuss these changes further.

### CODE REVIEW

File: app/models/event.rb [hunk 1]
```diff
@@ -14,4 +14,8 @@ class Event < ApplicationRecord
   validates :address,     presence: true
 
   scope :not_happened, -> { where('occurred_at > ?', Time.current.yesterday).order(occurred_at: :asc) }
+
+  def decorate(view_context)
```
Comment (vyper) [inline app/models/event.rb]: I do not like this method in model... 
Maybe... Is better change to controller?

File: app/controllers/events_controller.rb [hunk 2]
```diff
@@ -2,13 +2,19 @@ class EventsController < ApplicationController
...
-      format.html { @events = @events.decorate }
-      format.atom
+      format.html { @events = @events.map { |e| e.decorate(view_context) } }
+      format.atom { @markdown_renderer = markdown_renderer }
     end
   end
 
   def show
-    @event = Event.friendly.find(params[:id]).decorate
+    @event = Event.friendly.find(params[:id]).decorate(view_context)
```
Comment (vyper) [inline app/controllers/events_controller.rb]: Why not? 

```ruby
@event = EventDecorator.new(Event.friendly.find(params[:id]), view_context) 
```

File: app/decorators/event_decorator.rb [hunk 3]
```diff
@@ -1,55 +1,67 @@
-class EventDecorator < ApplicationDecorator
-  include Draper::LazyHelpers
+class EventDecorator < DelegateClass(Event)
```
Comment (vyper) [inline app/decorators/event_decorator.rb]: I didn't knew `DelegateClass` 😬

File: app/models/event.rb (hunk 1 above)
Comment (yez) [inline app/models/event.rb]: I think if we change the controller the way you suggested below, we can get rid of this entirely.

File: app/decorators/event_decorator.rb (hunk 3 above)
Comment (yez) [inline app/decorators/event_decorator.rb]: It is handy!

**Comment by yez** [comment]
@vyper I think that a Presenter would be fine but not all that different than what we have built here. I wanted to replace a Decorator with a Decorator to make the smallest possible change while removing Draper.

**Comment by vyper** [comment]
Thanks for you PR @yez! (:

**Pull request merged**